flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka09PartitionDiscoverer.java

```diff
@@ -74,7 +74,12 @@ protected void initializeConnections() {
 
 		try {
 			for (String topic : topics) {
-				for (PartitionInfo partitionInfo : kafkaConsumer.partitionsFor(topic)) {
+				List<PartitionInfo> topicPartitions = kafkaConsumer.partitionsFor(topic);
+				if (topicPartitions == null) {
+					throw new IllegalStateException("The topic " + topic + " does not exist");
```

Comment:
I fear that this might be too aggressive.
IMO, it is fine that the user has, say 3 topics, but only one of them actually doesn't exist.

What we should handle is the case where there is completely no partitions at all across all provided topics.
Perhaps for this, we only write a log that some topic has no partitions?

Reply:
Currently, it just throws NPE on running into a first non-existing topic.
Taking into account both your and Stephan's comment, I think the most consistent behavior could be to return the partitions that we're able to find. So, if no topics - then, an empty list.
But in case of an empty list, the AbstractPartitionDiscoverer's discoverPartitions() itself throws RuntimeException, so we get back to the initial situation. Is it ok?

Reply:
I think the `RuntimeException` in `AbstractPartitionDiscoverer#discoverPartitions` maybe needs to be revisited, also.
As far as I understand it, we should only fail the job if for the first discovery (for seed initial partitions that the connector consumes) is empty across all partitions. Otherwise, it should be ok that while the job runs, the partitions discover fails fetch partition meta info for some discovery attempt.

Reply:
I've took a look into it. Ideally we have to make the go/fail decision one level higher, at `FlinkKafkaConsumerBase`. It uses the partition discovery both for the initial seed and further on the run, so that's the guy who should decide. The problem I see here is that the `AbstractPartitionDiscoverer#discoverPartitions` doesn't just get the list of available partitions, but also filters the partitions applicable for the current task. So, once we get the partition list, we can't say whether the list is empty because nothing is found, or because the partitions have been post-filtered.
The only way to communicate this difference, as I see it now, is to introduce some new specific exception, and catch it at the `FlinkKafkaConsumerBase`.